### PR TITLE
feat(forms): back `MultiThumbSlider` with new `useSlider` hook

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Event and accessibility logic are now contained in `@zendeskgarden/container-slider`. This includes feature 🚀 support for the new `jump` prop that supports WAI-ARIA [page up/down keyboard navigation](https://www.w3.org/WAI/ARIA/apg/example-index/slider/slider-multithumb.html#kbd_label).

## Detail

Consideration was given to adding uncontrolled component support (offered by the `useSlider` hook), but without backing inputs, this makes no sense for the current component. Props for future consideration include:

- `defaultMinValue` and `defaultMaxValue` uncontrolled values connected to backing `<input hidden>` elements (see reverted 8e20ec4d05a77eb930d97672367844a5c06d1af9)
- `minValueText` and `maxValueText` for optional clarification of `aria-valuenow` (see https://www.w3.org/WAI/ARIA/apg/patterns/slidertwothumb/#wai-aria-roles-states-and-properties-18)

## Checklist

- [ ] :ok_hand: ~design updates will be Garden Designer approved (add the designer as a reviewer)~
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :guardsman: includes new unit tests
- [x] :wheelchair: tested for [WCAG 2.1](https://www.w3.org/TR/WCAG21) AA compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

File: packages/forms/src/elements/MultiThumbRange.tsx [hunk 1]
```diff
@@ -28,382 +19,101 @@ import {
...
+
 /**
  * @extends HTMLAttributes<HTMLDivElement>
  */
 export const MultiThumbRange = forwardRef<HTMLDivElement, IMultiThumbRangeProps>(
-  ({ min, max, minValue, maxValue, disabled, step, onChange, onMouseDown, ...props }, ref) => {
+  (
+    {
+      min = MIN,
+      max = MAX,
```
Comment (Francois-Esquire) [inline packages/forms/src/elements/MultiThumbRange.tsx:32]: Do we need to define a default value for `min` and `max` here if the `defaultProps` were set?

File: packages/forms/src/elements/MultiThumbRange.spec.tsx [hunk 2]
```diff
@@ -438,7 +456,7 @@ describe('MultiThumbRange', () => {
       const { getAllByTestId } = render(<MultiThumbRange minValue={50} maxValue={40} />);
       const thumb = getAllByTestId('thumb')[1];
 
-      expect(thumb).toHaveStyle('left: 50px');
+      expect(thumb).toHaveStyle('left: 40px');
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/forms/src/elements/MultiThumbRange.spec.tsx:459]: Yes, in the new hook we picked a consistent approach for applying out-of-bounds logic – that is, start-to-end (the minimum will determine where the maximum can start rather than the other way around). The previous logic was inconsistent in `MultiThumbRange`.

File: packages/forms/src/elements/MultiThumbRange.tsx (hunk 1 above)
Comment (jzempel) [reply packages/forms/src/elements/MultiThumbRange.tsx:32]: This helps avoid type-checked conditional logic for calculations that follow.

**Review by jgorfine-zendesk — CHANGES_REQUESTED** [review]
Hey @jzempel — overall, the code looks great; however, when I did some basic testing on the latest build, I ran into a couple of issues. 

## Dragging thumbs makes track fill disappear at ends of range

This happens both at the min and max:

https://user-images.githubusercontent.com/93289772/195139683-a2af2f76-b159-4c97-a352-b76c4058ad16.mov

## VoiceOver doesn't stay on thumb(s) when trying to increment or decrement with the arrow keys

I had observed similar behavior when I was working on the container originally, and I specifically noted that it went away with your version. Not totally sure what could be happening here:

https://user-images.githubusercontent.com/93289772/195140168-0c7b723f-d62f-4859-88c9-b33ea1ba3267.mov

Wondering if maybe it has to do with the DOM structure, now that it's inside of a Field?

File: packages/forms/src/types/index.ts [hunk 3]
```diff
@@ -159,12 +159,14 @@ export interface IMultiThumbRangeProps extends Omit<HTMLAttributes<HTMLDivElemen
   min?: number;
   /** Sets the maximum permitted value */
   max?: number;
-  /** Sets the minimum thumb input value */
+  /** Sets the minimum thumb value */
   minValue?: number;
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/forms/src/types/index.ts:163]: Not a bad thought. But given that we've never done this in the past, I'd rather not establish a new pattern without added consideration and intention.

**Comment by jzempel** [comment]
@jgorfine-zendesk

> Dragging thumbs makes track fill disappear at ends of range

I'm not able to reproduce. What are the characteristics of your environment. Also, do you have the same issues with the current https://zendeskgarden.github.io/react-components/?path=/story/packages-forms-multithumbrange--multi-thumb-range

> VoiceOver doesn't stay on thumb(s) when trying to increment or decrement with the arrow keys

Are you able to reproduce with https://zendeskgarden.github.io/react-components/?path=/story/packages-forms-multithumbrange--multi-thumb-range

If there are no regressions, we may want to pick these up as follow-on issues.

**Comment by jzempel** [comment]
@jgorfine-zendesk 

> VoiceOver doesn't stay on thumb(s) when trying to increment or decrement with the arrow keys

p.s. just checking... are you using the VO+arrow keys? For example, on this page – https://garden.zendesk.com/components/multi-thumb-range – I need opt+rightarrow in order to operate the slider when VO is active.

**Comment by jgorfine-zendesk** [comment]
> Are you able to reproduce with https://zendeskgarden.github.io/react-components/?path=/story/packages-forms-multithumbrange--multi-thumb-range

@jzempel I'll double-check and let you know. 👍 Totally agree that if this isn't something new, we shouldn't necessarily address it now.

> p.s. just checking... are you using the VO+arrow keys? For example, on this page – https://garden.zendesk.com/components/multi-thumb-range – I need opt+rightarrow in order to operate the slider when VO is active.

I was using the arrow keys by themselves. This is how I had tested the Slider container, too.

**Review by Francois-Esquire — APPROVED** [review]

**Comment by jgorfine-zendesk** [comment]
Hey @jzempel — following up on the two behaviors I flagged earlier this week. 

## Dragging thumbs makes track fill disappear at ends of range

Just tested the odd dragging behavior again on staging. Here's a fresh video of what I was seeing:

https://user-images.githubusercontent.com/93289772/195635502-4ffedf06-4306-4fe1-a273-7e944510c77b.mov

Interestingly enough, I experienced the same odd behavior on prod:

https://user-images.githubusercontent.com/93289772/195635591-bbe07c79-d5e6-41a0-85e8-a8da85abce9f.mov

However, I wasn't seeing this behavior at all on the Garden website, on prod:

https://user-images.githubusercontent.com/93289772/195635669-e1a8ef4e-f6ca-4e34-98bf-516c43752cb0.mov

This leads me to think that, maybe, the issue is with the Storybook demo itself. 

**tl;dr:** This probably is something we can revisit in the future. 

## VoiceOver doesn't stay on thumb(s) when trying to increment or decrement with the arrow keys

Just tested this again on staging —both with and without the VO keys. In this video, I'm interacting with the min value thumb with only the arrow keys, and then I'm interacting with the max value thumb with the arrow keys + VO keys:

https://user-images.githubusercontent.com/93289772/195636224-a43064a3-41aa-473d-892d-1bb59f01313b.mov

On prod, I was able to navigate the existing Multi-thumb range sample using only the arrow keys:

https://user-images.githubusercontent.com/93289772/195636374-58ff8b3a-fd55-4c80-9e6b-1618b03d5711.mov

Same for the Garden website, on prod:

https://user-images.githubusercontent.com/93289772/195636439-74b893ed-fc95-44b0-841f-e1a2d1368517.mov

This leads me to think that something is going on with the keyboard functionality code, or, perhaps, the DOM structure when the Multi-thumb range lives inside of a Field.

**tl;dr:** We may want to look into this a bit more now, as it potentially is a regression in functionality. 💥

**Comment by jgorfine-zendesk** [comment]
As a point of comparison, here's what I experienced when I tested the component on staging, via Assistiv Labs. In each of these tests, I had the screen reader on, and navigated using the left and right arrow keys only. 

## Chrome x JAWS (Windows)

https://user-images.githubusercontent.com/93289772/195640472-f2ef428c-b21e-4505-bcc1-845658a266ff.mov

## Chrome x NVDA (Windows)

https://user-images.githubusercontent.com/93289772/195642648-cfda45e4-086d-450f-a4a9-896c6c92d662.mov

**Comment by jzempel** [comment]
@jgorfine-zendesk I applied the 0.1.1 container fix and VO+arrow keys seem to be working as expected, on Safari, without screenreader drift.

**Review by jgorfine-zendesk — COMMENTED** [review]
Hey @jzempel, I have some good news and some bad news. 

## The good news

The Multi-thumb range appears to be working as intended on the following browser and screen reader combinations:

### VoiceOver x Safari

https://user-images.githubusercontent.com/93289772/196501894-30a57809-e206-4bd1-8a42-a225b77fc9a1.mov

### NVDA x Chrome (via AssistivLabs)

https://user-images.githubusercontent.com/93289772/196501961-b78af391-d77e-433a-a223-4ca668ded2f5.mov

### JAWS x Chrome (via AssistivLabs)

https://user-images.githubusercontent.com/93289772/196502002-52c26b48-cb32-4b1e-a5b2-ed1d901f49bc.mov

## The bad news

When I tested the staging instance on Chrome with VoiceOver on my local machine, I was completely unable to operate either slider with the keyboard —both with and without the VO keys. 

Here's a video in which the keyboard navigation is working just fine, without VoiceOver...and then becomes completely inoperable with it:

https://user-images.githubusercontent.com/93289772/196502313-7d576430-9ad5-45e4-b508-1ed459cc78ee.mov

## Moving forward

Since it's likely that someone using VoiceOver will be using it with Safari (per the [WebAIM Screen Reader Survey #9](https://webaim.org/projects/screenreadersurvey9/#browsercombos), conducted last year), and Safari + VoiceOver appears to be working as expected on both of our machines, I say we go ahead and merge these changes in. We could always loop in @zendesk-lzingarelli for additional testing and QA and do a fast followup if the bug rears its ugly head again.

**Comment by jzempel** [comment]
> When I tested the staging instance on Chrome with VoiceOver on my local machine, I was completely unable to operate either slider with the keyboard —both with and without the VO keys.

@jgorfine-zendesk at some point we might want to do a setup comparison. Chrome works fine for me with VO activated – as long as I hold down the `<option>` key.
